calico_node/startup/startup.go

```diff
@@ -295,8 +296,12 @@ func configureIPsAndSubnets(node *api.Node) {
 		adm := os.Getenv("IP6_AUTODETECTION_METHOD")
 		cidr := autoDetectCIDR(adm, 6)
 		if cidr != nil {
-			// We autodetected an IPv6 address so update the value in the node.
-			node.Spec.BGP.IPv6Address = cidr
+			// We autodetected an IPv6 address, if we have no previous IP, or it is different than the previous IP,
+			// we need to update it and check for conflicts.
+			if node.Spec.BGP.IPv6Address == nil || !node.Spec.BGP.IPv6Address.IP.Equal(cidr.IP) {
+				node.Spec.BGP.IPv6Address = cidr
+				checkConflictingNodes(client, node)
```

Comment:
Gah you're right, which kinda defeats the purpose, will refactor it to only call once